tests/PHPStan/Analyser/data/falsy-isset.php

```diff
@@ -82,11 +82,11 @@ public function maybeCertainNull(): void
 			assertType("array{bar: 1}", $a);
 		} else {
 			assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
-			assertType('array{bar?: null}', $a);
+			assertType('array{bar: null}', $a);
 		}
 
 		assertVariableCertainty(TrinaryLogic::createMaybe(), $a);
-		assertType("array{bar: 1}|array{bar?: null}", $a);
+		assertType("array{bar: 1}|array{bar: null}", $a);
```

Comment:
I would expect the type to be simplified to `array{bar|null}`

Reply:
I don't think so. `"array{bar: 1}|array{bar: null}"` only has 2 states, while `array{bar: 1|null}` has 3.